support-workers/src/main/scala/com/gu/emailservices/GuardianWeeklyEmailFields.scala

```diff
@@ -1,40 +1,43 @@
 package com.gu.emailservices
 
 import com.gu.emailservices.SubscriptionEmailFieldHelpers.formatDate
-import com.gu.support.catalog.FulfilmentOptions
+import com.gu.salesforce.Salesforce.SfContactId
+import com.gu.support.config.TouchPointEnvironment
 import com.gu.support.workers._
-import com.gu.support.workers.states.PaymentMethodWithSchedule
-import org.joda.time.LocalDate
+import com.gu.support.workers.states.ProductTypeCreated.GuardianWeeklyCreated
+import com.gu.support.zuora.api.ReaderType.{Direct, Gift}
 
 import scala.collection.immutable
+import scala.concurrent.{ExecutionContext, Future}
 
-object GuardianWeeklyEmailFields {
-  def build(
-    subscriptionEmailFields: SubscriptionEmailFields,
-    firstDeliveryDate: Option[LocalDate],
-    paymentMethodWithSchedule: PaymentMethodWithSchedule,
-    giftRecipient: Option[GiftRecipient] = None
-  ): EmailFields = {
-    import subscriptionEmailFields._
-    import allProductsEmailFields._
+class GuardianWeeklyEmailFields(
+  paperFieldsGenerator: PaperFieldsGenerator,
+  touchPointEnvironment: TouchPointEnvironment,
+  user: User,
+  sfContactId: SfContactId,
+) {
+  def build(gw: GuardianWeeklyCreated)(implicit ec: ExecutionContext): Future[EmailFields] = {
 
-    val additionalFields: immutable.Seq[(String, String)] = paymentMethodWithSchedule.paymentSchedule.payments.lift(1).map(
+    val additionalFields: immutable.Seq[(String, String)] = gw.purchaseInfo.paymentSchedule.payments.lift(1).map(
       payment => List("date_of_second_payment" -> formatDate(payment.date))
     ).getOrElse(Nil)
 
-    val fields: List[(String, String)] = PaperFieldsGenerator.fieldsFor(
-      subscriptionNumber,
-      billingPeriod,
-      user,
-      firstDeliveryDate,
-      currency,
-      paymentMethodWithSchedule,
-      directDebitMandateId,
-      promotion,
-      giftRecipient
-    ) ++ additionalFields
+    val giftRecipientFields = gw.giftRecipient.toList.flatMap(
+      recipient =>
+        List(
+          "giftee_first_name" -> recipient.firstName,
+          "giftee_last_name" -> recipient.lastName,
+        )
+    )
```

Comment:
for some reason gifts were being handled by the `PaperFieldsGenerator.fieldsFor` even though they only exist for GW.  I have moved it out here instead as the giftRecipient field no longer exists on paper subs.